raster/src/main/scala/geotrellis/raster/ArrayMultibandTile.scala

```diff
@@ -428,4 +428,25 @@ class ArrayMultibandTile(_bands: Array[Tile]) extends MultibandTile with MacroMu
 
     new ArrayMultibandTile(newBands)
   }
+
+  override def equals(other: Any) = {
+    if (other == null) {
+      false
+    } else other match {
+      case that : ArrayMultibandTile =>
+        case class NotEqualException() extends Exception()
+        var result = true
+        try {
+          for (b <- 0 until bandCount) {
+            if (band(b) != that.band(b)) throw NotEqualException()
+          }
+        } catch {
+          case e : NotEqualException =>
```

Comment:
I decided to use a while loop, the drawback in the new code is that the use of "return" keyword makes the code too imperative, but I think it's not an issue for such a small method, the imperativity will not spread out of its little sandbox :).